### PR TITLE
Tests for tasks and journals

### PR DESCRIPTION
**Pull request by tobixen**
Test for #97 - some of those tests are as of now intentionally broken, but at least 2/3 of the remaining tests should pass once #97 is fixed.

(I'm a bit uncertain on the recurring task with DUE date set but no DTSTART - RRULE is linked to DTSTART according to the RFC, so perhaps that ical data should be considered invalid - though, in my head one should consider the missing DTSTART to be equal to the DUE if no DTSTART is given).

### CODE REVIEW

**Comment by niccokunzmann** [comment]
> , in my head one should consider the missing DTSTART to be equal to the DUE if no DTSTART is given).

I think, I do the same with the events. They all have a DTEND. This module is about ease and consistency, so people do not need to worry much about the edge cases.

File: recurring_ical_events.py [hunk 1]
```diff
@@ -360,6 +360,10 @@ def __init__(self, calendar, keep_recurrence_attributes=False):
         self.repetitions = []
         for event in calendar.walk("VEVENT"):
             self.repetitions.append(RepeatedEvent(event, keep_recurrence_attributes))
+        for event in calendar.walk("VTODO"):
+            self.repetitions.append(RepeatedTodo(event, keep_recurrence_attributes))
+        for event in calendar.walk("VJOURNAL"):
```
Comment (niccokunzmann) [inline recurring_ical_events.py:365]: Does `walk(("VEVENT", "VTODO"))` work? - not a blocker for merge

Comment (tobixen) [reply recurring_ical_events.py:365]: Oh, did I really commit this?  It's not part of the "broken tests" and it's anyway broken.  I'll move it over to a separate pull request.

Comment (tobixen) [reply recurring_ical_events.py]: > Does `walk(("VEVENT", "VTODO"))` work? - not a blocker for merge

Apparently not:

```
(Pdb) self.walk(('VEVENT', 'VTODO'))
*** AttributeError: 'tuple' object has no attribute 'upper'
```

**Comment by niccokunzmann** [comment]
Can this be closed now? What do you think?

**Comment by tobixen** [comment]
Probably.  Would be nice with more test code.  At least, I'd like to run the caldav test suite again to see if anything is amiss.

**Comment by tobixen** [comment]
aww ...

```
    def between(self, after, before, inc=False, count=1):
        """ Returns all the occurrences of the rrule between after and before.
        The inc keyword defines what happens if after and/or before are
        themselves occurrences. With inc=True, they will be included in the
        list, if they are found in the recurrence set. """
        if self._cache_complete:
            gen = self._cache
        else:
            gen = self
        started = False
        l = []
        if inc:
            for i in gen:
>               if i > before:
E               TypeError: '>' not supported between instances of 'datetime.datetime' and 'NoneType'

/usr/lib/python3.10/site-packages/dateutil/rrule.py:284: TypeError
```

**Comment by tobixen** [comment]
(nevermind my previous message, that was a bug at my side)

**Comment by tobixen** [comment]
The tests at my side still doesn't pass (I thought I had checked that ... but perhaps not after the last changes).

I need to sleep.  Hope to get time to look more into it tomorrow.

**Comment by niccokunzmann** [comment]
Yes. Sleep helps :) It is nice it we can reproduce the behavior here. If you like, you can open issues whenever you encounter something.

**Comment by tobixen** [comment]
It helped sleeping.  For one thing, this is a pull request which was fully included in pull request #100, so of course it can be closed.  For the second thing, I managed to get all tests passing at my side.

**Comment by niccokunzmann** [comment]
Woohoo. Celebration!

**Pull request closed without merging**